### PR TITLE
New Job methods for updating custom job data.

### PR DESCRIPTION
**Pull request by phbcanada**
New specific methods for saving a Tray ID and a list of Mesh filenames in the job custom data.

### CODE REVIEW

**Comment by phbcanada** [comment]
@gregra81 - please merge if this looks ok.
Thanks.

**Comment by gregra81** [comment]
@rozman1234, can took a look?

File: sdk/src/Job.js [hunk 1]
```diff
@@ -270,7 +270,111 @@ var ADSKSpark = ADSKSpark || {};
...
+         * set value for this field.
+         * @param {String} [trayId] - The Spark Id of the Tray used to create the printable
+         * for this job.
+         * @returns {Promise} - A Promise which resolves to this object with updated contents.
+         */
+        setSourceTray: function (trayId) {
+            if (!this.id) {
+                return Promise.reject(new Error('Unknown job in setSourceTray.'));
+            }
+            if (!this.data.job_custom_data)
```
Comment (rozman1234) [inline sdk/src/Job.js]: Please add curly brackets in all IF conditions

**Comment by phbcanada** [comment]
@rozman1234 - done.

**Pull request merged**